packages/blade/src/components/Box/BaseBox/types.ts

```diff
@@ -1,44 +1,155 @@
 import type { CSSObject } from 'styled-components';
-import type { DotNotationSpacingStringToken } from '~src/_helpers/types';
-import type { Border } from '~tokens/global';
-
-export type BaseBoxProps = {
-  display?: CSSObject['display'];
-  flex?: CSSObject['flex'];
-  flexWrap?: CSSObject['flexWrap'];
-  flexDirection?: CSSObject['flexDirection'];
-  flexGrow?: CSSObject['flexGrow'];
-  flexShrink?: CSSObject['flexShrink'];
-  alignItems?: CSSObject['alignItems'];
-  alignContent?: CSSObject['alignContent'];
-  justifyContent?: CSSObject['justifyContent'];
-  alignSelf?: CSSObject['alignSelf'];
-  overflow?: CSSObject['overflow'];
-  position?: CSSObject['position'];
-  top?: CSSObject['top'];
-  right?: CSSObject['right'];
-  bottom?: CSSObject['bottom'];
-  left?: CSSObject['left'];
-  paddingTop?: DotNotationSpacingStringToken | 'auto';
-  paddingBottom?: DotNotationSpacingStringToken | 'auto';
-  paddingLeft?: DotNotationSpacingStringToken | 'auto';
-  paddingRight?: DotNotationSpacingStringToken | 'auto';
-  marginTop?: DotNotationSpacingStringToken | 'auto' | number;
-  marginBottom?: DotNotationSpacingStringToken | 'auto' | number;
-  marginLeft?: DotNotationSpacingStringToken | 'auto' | number;
-  marginRight?: DotNotationSpacingStringToken | 'auto' | number;
-  borderRadius?: keyof Border['radius'];
-  width?: CSSObject['width'];
-  height?: CSSObject['height'];
-  gap?: DotNotationSpacingStringToken | 'auto' | number;
-  minHeight?: number;
-  minWidth?: number;
-  maxHeight?: number;
-  maxWidth?: number;
-  className?: string;
-  transform?: CSSObject['transform'];
-  background?: CSSObject['background'];
-  backgroundColor?: CSSObject['backgroundColor'];
-  zIndex?: CSSObject['zIndex'];
-  id?: string;
+import type { Theme } from '~components/BladeProvider';
+import type { Border, Spacing } from '~tokens/global';
+import type { Breakpoints } from '~tokens/global/breakpoints';
+import type { DotNotationColorStringToken } from '~src/_helpers/types';
+
+type MakeValueResponsive<T> =
+  | T
+  | {
+      // Using this instead of Record to maintain the jsdoc from breakpoints.ts
+      [P in keyof Breakpoints]?: T;
+    };
+type MakeObjectResponsive<T> = { [P in keyof T]: MakeValueResponsive<T[P]> };
+type ArrayOfMaxLength4<T> = readonly [T?, T?, T?, T?];
+
+type SpaceUnits = 'px' | 'fr' | '%' | 'rem' | 'em';
+type SpacingValueType = `spacing.${keyof Spacing}` | `${string}${SpaceUnits}` | 'auto';
```

Comment:
`fr` works only for grid properties so providing the same typing for all properties would be bit inconvenient.

Reply:
It anyway doesn't show up in autocomplete so should be fine I guess? its like string only except it throws error when units are not added. I added it because someone might do `marginBottom="12"` which has same issues as `marginBottom={12}` so better to not allow just string?

Reply:
Can't think of an edge case where this might not work. Let's go ahead, we can always make it `string` later if required.